internal/testintegration/integration_test.go

```diff
@@ -32,33 +32,43 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// defaultClient holds the k8s client for the default EnvTestHarness which is
+// shared among tests that do not mess with the manager's lifecycle.
+var defaultClient client.Client
+
 func TestMain(m *testing.M) {
-	teardown, err := testintegration.EnvTestSetup()
+	// start up a shared EnvTestHarness to be reused across test cases that do not
+	// impact the lifecycle of the manager. This makes tests cases more efficient
+	// because it takes 2-3 minutes to start up a new EnvTestHarness.
+	var err error
+	th, err := testintegration.EnvTestSetup()
```

Comment:
Shall we call this constructor `NewTestHarness()`?

Reply:
Or `NewTestEnv`? We should settle on a name and make the variable names match (e.g. `th` for test harness implies `NewTestHarness` etc).

Reply:
Renamed NewTestHarness()